src/Texture.cpp

```diff
@@ -0,0 +1,49 @@
+#include "Texture.h"
+
+namespace ccm
+{
+	Texture::Texture(std::string_view textureSource) : m_texture(nullptr, SDL_DestroyTexture)
+	{
+		auto[pTex, width, height, pitch] = TextureGenerator::loadTextureFromFile(textureSource);
+		m_texture.reset(pTex);
+		m_width = width;
+		m_height = height;
+		m_pitch = pitch;
+	}
+
+	Texture::Texture(Texture&& other)
```

Comment:
The move constructor and assignment looks like the `= default` behavior. (default does member-wise moves)